examples/cluster-go/main.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"github.com/pulumi/pulumi-eks/go/eks"
```

Comment:
So, one thing I've been wondering...

All our other providers put the SDK files under `<repo>/sdk`, e.g.:

- `pulumi/pulumi-aws/sdk/dotnet`
- `pulumi/pulumi-aws/sdk/go`
- `pulumi/pulumi-aws/sdk/nodejs`
- `pulumi/pulumi-aws/sdk/python`

However, since this repo (eks) was originally only Node.js, it didn't have an `sdk` dir -- all the Node.js files were in a `nodejs` dir in the root fo the repo (`pulumi/pulumi-eks/nodejs`).

When I went to add Python and .NET, I contemplated moving the existing Node.js files under a new `sdk` dir, but ended up holding off and instead creating `python` and `dotnet` dirs in the root of the repo (with the idea that I'd eventually go back and clean this up). In hindsight, I probably should have just gone ahead and created the `sdk` dir and moved all the sdk files under it.

The structure of the repo doesn't matter much for using these language SDKs, but with Go, you're importing paths in the repo, and all our other repos put the go files under `sdk`.

I'm wondering if we should just go ahead and do that for the Go SDK now, before anyone has taken a dependency on the import path.

I think this means reworking the existing go.mod/sum in the root of the repo, something like:

- Add `sdk` dir
- Generate Go files under `sdk/go`.
- Remove `go.mod/sum` in repo root.
- Add `go.mod/sum` in `sdk` dir (with just dependencies needed for the SDK)
- Add `go.mod/sum` under `examples` (with just dependencies needed to run the tests)
- Move `utils/utils.go` to `examples/utils/utils.go`
- Add `go.mod/sum` under `provider` (with just dependencies necessary for the provider and code generator program)

Ideally, we'd then also move the `nodejs`, `python`, and `dotnet` dirs under `sdk`, but we can do that later, as a separate change (will need to ensure all the paths are fixed up across the tests, GHA, etc.).

Reply:
I agree that we should move Go at least to live in the sdk directory since codegen has that assumption baked in several places (e.g., the PkgVersion function regex).

Reply:
Sounds good. Will try this out and update the PR shortly.

Reply:
I updated things as you recommended @justinvp. I haven't quite run the tests yet after the reorg but things build. Will look to add a simple starting go test integration test wrapper as well.